packages/big-design/src/components/Worksheet/Cell/Cell.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { memo, useCallback, useEffect, useMemo } from 'react';
+
+import { Small } from '../../Typography';
+import { useEditableCell, useStore } from '../hooks';
+import { Cell as TCell } from '../types';
+
+import { StyledCell } from './styled';
+
+interface CellProps extends TCell<string | number> {
+  validation?(value: string | number): boolean;
+}
+
+export const Cell: React.FC<CellProps> = memo(({ columnIndex, hash, rowIndex, type, value, validation }) => {
+  const cell = useMemo(() => ({ columnIndex, rowIndex, hash, type, value }), [
+    columnIndex,
+    rowIndex,
+    hash,
+    type,
+    value,
+  ]);
+
+  const { handleDoubleClick, handleBlur, handleKeyDown, isEditing, Editor } = useEditableCell(cell);
+  const setSelectedRows = useStore(useMemo(() => (state) => state.setSelectedRows, []));
+  const setSelectedCells = useStore(useMemo(() => (state) => state.setSelectedCells, []));
+  const addInvalidCells = useStore(useMemo(() => (state) => state.addInvalidCells, []));
+  const removeInvalidCells = useStore(useMemo(() => (state) => state.removeInvalidCells, []));
+
+  const isSelected = useStore(
+    useMemo(
+      () => (state) =>
+        state.selectedCells.reduce(
```

Comment:
I like this use of reduce, but jw is there a reason not to use `Array.some()` here instead?

```suggestion
        state.selectedCells.some(selectedCell => selectedCell.columnIndex === cell.columnIndex && selectedCell.rowIndex === cell.rowIndex)
```

Reply:
Good call